src/fondant/pipeline/compiler.py

```diff
@@ -487,3 +487,133 @@ def _set_configuration(task, fondant_component_operation):
             task.set_accelerator_type(accelerator_name)
 
         return task
+
+
+class SagemakerCompiler(Compiler):
+    def __init__(self):
+        self._resolve_imports()
+
+    def _resolve_imports(self):
+        try:
+            import sagemaker
+            import sagemaker.processing
```

Comment:
do we need all those extra imports for checking? I think `import sagemaker` might suffice

Reply:
oh ok good to know

Reply:
Can you test it to be sure? :)

Reply:
```
  File "/Users/georgeslorre/ML6/internal/express/src/fondant/pipeline/compiler.py", line 594, in compile
    step = self.sagemaker.workflow.steps.ProcessingStep(
AttributeError: module 'sagemaker.workflow' has no attribute 'steps'
```

when I comment out the 3 extra imports